### PR TITLE
Add unstable Python 3.12 to CI allowing failures

### PR DESCRIPTION
**Pull request by webknjaz**
Ref https://github.com/python-attrs/attrs/pull/1029#issuecomment-1261394538.

This is kept as a draft that is expected to be merged once all of the following are true:
* [x] Python 3.11 final is out
* [x] GitHub updated their available CPython builds to include Python 3.12-dev
* [x] The DNM (do-not-merge) tag is dropped from the PR title

### CODE REVIEW

**Comment by webknjaz** [comment]
> * [ ]  Python 3.11 final is out
> * [ ] GitHub updated their available CPython builds to include Python 3.12-dev

So the new job is failing in 6 seconds, which means that it's also okay(ish) to merge this now — it won't waste too much of the CI resources. But there would be a downside of having a red mark in some places on the UI, which is going to be annoying.

**Comment by hynek** [comment]
Yeah not merging until it's passing.

Also lol you've introduced yourself a conflict. ;)

**Comment by webknjaz** [comment]
> Also lol you've introduced yourself a conflict. ;)

Yep, I was expecting that. That's a side effect of having a habit to keep commits and PRs atomic :man_shrugging:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,9 +34,13 @@ jobs:
           - "3.9"
           - "3.10"
           - "~3.11.0-0"
```
Comment (webknjaz) [inline .github/workflows/ci.yml:36]: @hynek note that with this, 3.11 is also allowed to fail. The conditional may need to be improved if you don't want that.

**Comment by webknjaz** [comment]
@hynek looks like 3.12 alpha is available now

**Comment by webknjaz** [comment]
I suppose coverage might've dropped independently.

**Comment by webknjaz** [comment]
> I'm guessing coverage.py isn't there yet

Looks like you're right.

**Comment by webknjaz** [comment]
@hynek it occurred to me that maybe the jobs that are allowed to fail shouldn't influence the coverage check? WDYT?

**Comment by webknjaz** [comment]
@hynek it's green now!

**Review by hynek — APPROVED** [review]

**Pull request merged**